lib/biokbase/catalog/db.py

```diff
@@ -164,11 +164,19 @@ def _initialize_mongo_client(self):
             if self.mongo_user and self.mongo_psswd:
                 # Connection string with authentication
                 mongo_client = MongoClient(
-                    f"mongodb://{self.mongo_user}:{self.mongo_psswd}@{self.mongo_host}/{self.mongo_db}?authMechanism={self.mongo_authMechanism}"
+                    self.mongo_host,
+                    username=self.mongo_user,
+                    password=self.mongo_psswd,
+                    authSource=self.mongo_db,
+                    authMechanism=self.mongo_auth_mechanism,
+                    retryWrites=self.mongo_retry_writes
```

Comment:
How did you test this was set correctly?

Reply:
How about a full test from starting the service with and without it?

Reply:
tests added

Reply:
Oof, we miscommunicated here. I was talking about manually making sure that the retry write setting is transmitted correctly from wherever it's set (e.g. in rancher, for example) to the db class. The reason I'm asking about this is because it obviously couldn't have worked before due to https://github.com/kbase/catalog/pull/146#discussion_r1982315042

Reply:
I tested it rancher. 

<img width="960" alt="Screenshot 2025-03-07 at 2 03 59 PM" src="https://github.com/user-attachments/assets/3d0bd281-614e-4c5d-a53d-3e4b7d3eab0b" />

Reply:
:+1: